CRM/Utils/Check/Component/Smarty.php

```diff
@@ -25,23 +25,24 @@ class CRM_Utils_Check_Component_Smarty extends CRM_Utils_Check_Component {
    */
   public function checkSmartyVersion(): array {
     $messages = [];
-    $smarty2Path = \Civi::paths()->getPath('[civicrm.packages]/Smarty/Smarty.class.php');
-    $path = CRM_Utils_Constant::value('CIVICRM_SMARTY_AUTOLOAD_PATH') ?: CRM_Utils_Constant::value('CIVICRM_SMARTY3_AUTOLOAD_PATH');
-    if ($path === $smarty2Path) {
-      $smartyPath = \Civi::paths()->getPath('[civicrm.packages]/smarty5/Smarty.php');
-
+    $settingNames = ['CIVICRM_SMARTY_AUTOLOAD_PATH', 'CIVICRM_SMARTY3_AUTOLOAD_PATH'];
+    foreach ($settingNames as $settingName) {
+      if (CRM_Utils_Constant::value($settingName)) {
+        $pathSetting = CRM_Utils_Constant::value($settingName);
+        break;
+      }
+    }
+    if ($pathSetting && !str_ends_with($pathSetting, 'smarty5/Smarty.php')) {
```

Comment:
I wonder if we should add an elseif onto this and asy if you have one of thse defined and it is smarty5 you should remove the define as it is not doing anything?

Reply:
@seamuslee001 we could but that would be a different message, probably a notice rather than an error.